artemis-core-client/src/main/java/org/apache/activemq/artemis/spi/core/protocol/RemotingConnection.java

```diff
@@ -35,9 +35,11 @@
 /**
  * A RemotingConnection is a connection between a client and a server.
  *
- *
  * Perhaps a better name for this class now would be ProtocolConnection as this
  * represents the link with the used protocol
+ *
+ * A default noop implementation is provided. This, for example, makes test
+ * implementations much simpler.
```

Comment:
Though they can also make managing multiple 'real implementations' as the codebase has a little more awkward, and can have an overhead vs concrete method impl. The interface becomes less streamlined also.

There only looks to be 1 test impl this actually changed (which could perhaps have just used mocking instead, maybe using even less lines than the defaultings) and then also 1 'fake' management connection. I'd personally use the abstract one as  proper base for all the real impls like you did and leave the management one closer to as-is than convert all the interface methods to default to largely return false/null just so the fake one doesnt have to.

Reply:
Nice call. I implemented your suggestion.